plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -732,7 +686,10 @@ func getUsedPorts(pods ...*api.Pod) map[int]bool {
 	for _, pod := range pods {
 		for _, container := range pod.Spec.Containers {
 			for _, podPort := range container.Ports {
-				ports[podPort.HostPort] = true
+				// TODO: Aggregate it at the NodeInfo level.
+				if podPort.HostPort != 0 {
```

Comment:
This gives us significant performance gains.
To clarify, currently for every container.Ports in each container, we are adding a HostPort to the "ports" result. However, if you don't have HostPort defined, it is 0 and "0" is explicitly ignored in PodFitsHostPorts which is the only function using this one.

However, since we were returning "0" in the result previously, we didn't hit an optimization of:
if len(wantPorts) == 0 {
  return true, nil
}
so we were computing ports from all pods on that node.
Which made it significantly more expensive. 

Now we simply don't do it.